### DISCUSSION TITLE
AdGuard Home & pixelsrv-tls

### DISCUSSION THREAD

**Discussion by gspannu** [AdguardTeam/AdGuardHome#3949] [Q&A]
**Is there any guide for installing Pixelsrv along with AdGuardHome?**

There are articles/ links that discuss how to configure AdGuard Home to use pixelsrv, but _I cannot find anything that provides a guide to install pixelsrv_.

My setup is as follows:
- Debian 11 running on a VPS (in the cloud)
- AdGuard Home (Beta version) running on the Debian server
- AdGuard Home encryption enabled, https enabled. (I am using Lets Encrypt certificates for encryption, generated using certbot utility)
- AdGuard Home providing plain, DoT and DoH to connecting clients
- DoT on port 783, DoH on port 443, Plain DNS on port 53
- All others ports blocked via firewall

**Questions:**
- How do I install pixelsrv?
- Any **guides/ help/ links** to setup pixelsrv properly, so that it replaces both http and https ads with single pixel?

Objective:
Removing extra blank spaces in web pages.

**Comment by ainar-g** [comment]
Is the wiki page that mentions `pixelsrv-tls` not informative enough?

https://github.com/AdguardTeam/AdGuardHome/wiki/FAQ#customblock

**Reply by gspannu** [reply]
The Wiki page talks about setting up AdGuard Home **_after_ Pixelsrv-tls has been installed.** 

My query as written above is ‘_how to/ instructions to install pixelate-tls_’ itself on a Debian installation. 
These are not part of the Wiki (and shouldn’t be either). 

There isn’t much help on the pixelsrv page either. 

Hence, I am looking for help from **someone who has installed pixelsrv on a Debian server**.

**Comment by gspannu** [comment]
Anyone here installed pixelsrv on a Debian machine that is also running AdGuardHome?

any help would be much appreciated?
